### PR TITLE
chore(processor): Introduce typed envelope for processing

### PR DESCRIPTION
**Pull request by olksdr**
This change adds a small typed wrapper for managed envelope which is used in the processor.
Also add a specific marker trait to tighten restriction for function which used only for event based processing. 

Also, 
- remove the types from within the `ProcessingGroup` enum variants, they are not used and it's not necessary for now
- ensures that the envelope can be stored or sent to upstream only if it's processed in the processor
 
There is no functional changes in this PR.

related: https://github.com/getsentry/team-ingest/issues/253

#skip-changelog

### CODE REVIEW

File: relay-server/src/services/processor.rs [hunk 1]
```diff
@@ -147,38 +157,44 @@ pub struct ForwardUnknownGroup;
...
 /// Describes the groups of the processable items.
 #[derive(Clone, Copy, Debug)]
 pub enum ProcessingGroup {
     /// All the transaction related items.
     ///
     /// Includes transactions, related attachments, profiles.
-    Transaction(TransactionGroup),
-    /// All the items which require (have or create) events.
+    Transaction,
+    /// All the items which require g events.
```
Comment (jjbayer) [inline relay-server/src/services/processor.rs]: False positive?

File: relay-server/src/services/processor.rs [hunk 2]
```diff
@@ -147,38 +157,44 @@ pub struct ForwardUnknownGroup;
...
-    CheckIn(CheckInGroup),
+    CheckIn,
     /// Spans.
-    Span(SpanGroup),
+    Span,
     /// Metrics.
-    Metrics(MetricsGroup),
-    /// Unknown item types will be forwarded upstream (to processing Relay), where we will
+    Metrics,
+    /// Unknown item types will be forwarded upstream g, where we will
```
Comment (jjbayer) [inline relay-server/src/services/processor.rs]: False positive?

**Review by jjbayer — APPROVED** [review]

**Review by Dav1dde — APPROVED** [review]

File: relay-server/src/services/processor.rs (hunk 1 above)
Comment (olksdr) [reply relay-server/src/services/processor.rs]: Yeah, find/replace didn't properly workout 😬  
Thanks!

File: relay-server/src/utils/managed_envelope.rs [hunk 3]
```diff
@@ -65,6 +68,49 @@ struct EnvelopeContext {
     group: ProcessingGroup,
 }
 
+/// A wraper for [`ManagedEnvelope`] with assigned processing group type.
```
Comment (TBS1996) [inline relay-server/src/utils/managed_envelope.rs:71]:
```suggestion
/// A wrapper for [`ManagedEnvelope`] with assigned processing group type.
```

**Review by TBS1996 — APPROVED** [review]

**Pull request merged**